app/models/article.rb

```diff
@@ -73,4 +77,10 @@ def generate_published_dates
   def downcase_content_format
     self.content_format = self.content_format.downcase
   end
+
+  def create_redirect
+    unless Redirect.exists?(source_path: short_path, target_path: path)
+      Redirect.create(source_path: short_path, target_path: path)
+    end
+  end
```

Comment:
@greggawatt Do you have an ETA for landing this in master? There is probably a fair bit of rebasing needed to catch up to the latest.